infrastructure/provider/kavenegar.go

```diff
@@ -0,0 +1,39 @@
+package provider
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type KavenegarProvider struct {
+	urlFormat string
+	token     string
+}
+
+func (s *KavenegarProvider) makeRequestUrl(sender, receiver, message string) string {
+	url := fmt.Sprintf(s.urlFormat, s.token, receiver, sender, message)
+	return url
+
+}
+
+func (s *KavenegarProvider) makeReceivers(receiver []string) string {
+	strReceivers := strings.Join(receiver, ",")
+	return strReceivers
+
+}
+
+func (s *KavenegarProvider) SendMessage(sender, msg string, receivers interface{}) (isSuccessful bool) {
```

Comment:
instead of using interface{} use []string, since we know numbers are going to be a []string theres no need to use type assertion via interface{} and all the efficiency and security issues that come with it.

-- this happens in multiple places fix them all please.